backend/api/Areas/Insurances/Controllers/InsuranceController.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using MapsterMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Pims.Api.Areas.Insurance.Models;
+using Pims.Api.Policies;
+using Pims.Dal;
+using Pims.Dal.Security;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace Pims.Api.Areas.Insurance.Controllers
+{
+    /// <summary>
+    /// InsuranceController class, provides endpoints for interacting with insurances.
+    /// </summary>
+    [Authorize]
+    [ApiController]
+    [ApiVersion("1.0")]
+    [Area("insurances")]
+    [Route("v{version:apiVersion}/leases/{leaseId}/[area]")]
+    [Route("/leases/{leaseId}/[area]")]
+    public class InsuranceController : ControllerBase
+    {
+        #region Variables
+        private readonly IPimsService _pimsService;
+        private readonly IMapper _mapper;
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a InsuranceController class, initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="pimsService"></param>
+        /// <param name="mapper"></param>
+        ///
+        public InsuranceController(IPimsService pimsService, IMapper mapper)
+        {
+            _pimsService = pimsService;
+            _mapper = mapper;
+        }
+        #endregion
+
+        #region Endpoints
+        /// <summary>
+        /// Updates a list of insurances for an insurance.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost()]
+        [HasPermission(Permissions.LeaseEdit)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(Api.Models.BatchUpdate.Reply<InsuranceModel>), 200)]
+        [SwaggerOperation(Tags = new[] { "insurance" })]
+        public IActionResult BatchUpdate(int leaseId, Api.Models.BatchUpdate.Request<InsuranceModel> batchUpdate)
```

Comment:
We can take this as it is for now, but I have some scenarios that I worry about that would be solved by checking the concurrency on the lease:

1) concurrent adds. If user 1 adds insurance A, and then user 2 adds insurance A' (equivalent to A) If the backend doesn't enforce the concurrency id of the lease the system will end up with two of the "same" leases.
2) concurrent delete/updates If user 1 deletes insurance A, and then user 2 updates insurance A, what will EFs update method do? update dynamically adds new and updates existing entities. Hopefully because you should be passing an id ef will throw an error instead of adding it?

Reply:
That's a fair point. In the case of insurances, I don't think that is a problem since afaik there can be only one type of insurance per lease, so duplicates would not be allowed.